### PR TITLE
Move modal from head to footer

### LINKED ISSUE #25434: Blockwishlist module adds modals into <head>
#### Describe the bug

The html markup generated by blockwishlist is invalid. It adds the html code of the models using `hookDisplayHeader`, thus they are ending up inside the `<head>` tag instead of the `<body>`. 

#### Expected behavior

The generated HTML code should be valid.

#### Steps to Reproduce

1. With blockwishlist installed look at the source code. You will see:

```
<head>
(...) // alot of stuff
<!-- begin module:blockwishlist/views/templates/hook/displayHeader.tpl -->
<!-- begin /modules/blockwishlist/views/templates/hook/displayHeader.tpl -->
<!-- begin /home/www-data/ps/test1/releases/20210727080251/modules/blockwishlist/views/templates/components/modals/add-to-wishlist.tpl -->
<div
  class="wishlist-add-to"
  data-url="https://ch.test1.ps.sbgmb.net/module/blockwishlist/action?action=getAllWishlist"
> //                              <=  <div> is invalid here in <head>
(...) 
<!-- end /modules/blockwishlist/views/templates/hook/displayHeader.tpl -->
(...) // maybe other modules
</head>
<body id="index" class="lang-fr country-ch currency-chf layout-full-width page-index tax-display-enabled">
```

#### Additional information

* PrestaShop version: 1.7.7, blockwishlist 2.0.1
* PHP version: N/A

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | The modal is put to the header section with this PR it's in the footer
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/25434
| How to test?      | For this test you need to update the version of the module and made an update. After you need to check if module work and the modal is not on the header.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.
| Sponsor company   | @Touxten

### CODE REVIEW

**Review by NeOMakinG — APPROVED** [review]

File: upgrade/upgrade-2.0.2.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * @param BlockWishList $moduleadd upgrade
+ *
+ * @return bool
+ */
+function upgrade_module_2_0_2($module)
+{
+    return $module->registerHook('DisplayFooter') &&
+        $module->unregisterHook('DisplayHeader');
```
Comment (PierreRambaud) [inline upgrade/upgrade-2.0.2.php:30]:
```suggestion
    return $module->registerHook('displayFooter') &&
        $module->unregisterHook('displayHeader');
```

**Review by PierreRambaud — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @okom3pom ,

Sorry for the delay in the repsonse.
I didn't manage to see the fix.
I did the following :
- install module blockwishlist
- install PR #132 
- I could'nt upgrade to 2.0.2, v2.0.2 wasn't available.

Am I missing something ? ^^
Thanks!

**Comment by NeOMakinG** [comment]
@florine2623 I think you just have to install the module using this PRs branch, and it just should work as usually!

**Comment by florine2623** [comment]
Hello @okom3pom ,

Thanks for the PR !
The block is moved to the body : OK
But I'm facing an issue, I can't add any product to my wishlist with this PR.
Here's my screenrecord : https://watch.screencastify.com/v/A19M9Fv0WRIBDv5ieYOT

Without the PR it is OK :
https://watch.screencastify.com/v/NRuaedCcll5viMCqlZY7

Could you take a look ?
Thanks!

**Comment by okom3pom** [comment]
Hello @florine2623 

The update is to test if the hook is added, if you don't run update module don't work. 

1. install module blockwishlist
2. install PR Move modal from head to footer #132
3. Change the version number to  2.0.2 ( [ici](https://github.com/PrestaShop/blockwishlist/blob/dev/blockwishlist.php#L72) )
4. And makes an update of the module

Or you can just fork my branch and make a new install

Possible it's a DEV QA ?

**Comment by NeOMakinG** [comment]
@florine2623 @okom3pom I can do the QA if needed

**Comment by florine2623** [comment]
Hello @okom3pom @NeOMakinG ,

Thanks for the details :)

It is all good to me ! QA approved ✅

**Comment by Progi1984** [comment]
TThanks @florine2623 & @okom3pom

**Pull request merged**